.configurations/configuration.dsc.yaml

```
@@ -0,0 +1,96 @@
+# yaml-language-server: $schema=https://aka.ms/configuration-dsc-schema/0.2
+
+properties:
+  resources:
+    ########################################################################
+    # Section: Install VS-Code
+    ########################################################################
+    - resource: Microsoft.WinGet.DSC/WinGetPackage
+      id: install-vs-code
+      directives:
+        description: Install Microsoft Visual Studio Code
+        allowPrerelease: true
+      settings:
+        id: Microsoft.VisualStudioCode
+        source: winget
+        Ensure: Present
+    - resource: vscode/VSCodeExtension
+      dependsOn:
+        - install-vs-code
+      directives:
+        description: Install YAML extension for VSCode
+        allowPrerelease: true
+      settings:
+        Name: redhat.vscode-yaml
+        Ensure: Present
+    - resource: vscode/VSCodeExtension
+      dependsOn:
+        - install-vs-code
+      directives:
+        description: Install EditorConfig extension for VSCode
+        allowPrerelease: true
+      settings:
+        Name: EditorConfig.EditorConfig
+        Ensure: Present
+    - resource: vscode/VSCodeExtension
+      dependsOn:
+        - install-vs-code
+      directives:
+        description: Install PowerShell extension for VSCode
+        allowPrerelease: true
+      settings:
+        Name: ms-vscode.powershell
+        Ensure: Present
+    ########################################################################
+    # Section: Install Packages
+    ########################################################################
+    - resource: Microsoft.WinGet.DSC/WinGetPackage
+      id: install-git
+      directives:
+        description: Install Git # Although the user probably already has git installed, it's possible that they don't
+        allowPrerelease: true
+      settings:
+        id: Git.Git
+        source: winget
+        Ensure: Present
+    - resource: Microsoft.WinGet.DSC/WinGetPackage
+      id: install-git-cli
+      directives:
+        description: Install GitHub CLI
+        allowPrerelease: true
```

```suggestion
        allowPrerelease: true
        securityContext: elevated
```

If you add the security context here, users who are members of local administrators who need to elevate will get prompted right after the agreement dialog rather than later in the configuration if they have the configureSelfElevate enabled. It might also make sense to put this one first just to avoid that delay in the UAC.